server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMessageListMethodTest.java

```diff
@@ -1332,6 +1334,228 @@ public void getMessageListShouldComputeTextBodyWhenNoTextBodyButHtmlBody() throw
         ;
     }
 
+    @Test
+    public void getMessageListHasKeywordFilterShouldWork() throws Exception {
+        mailboxProbe.createMailbox(MailboxConstants.USER_NAMESPACE, username, "mailbox");
+
+        ComposedMessageId messageNotFlagged = mailboxProbe.appendMessage(username, new MailboxPath(MailboxConstants.USER_NAMESPACE, username, "mailbox"),
+            new ByteArrayInputStream("Subject: test\r\n\r\ntestmail".getBytes()), new Date(), false, new Flags());
+        ComposedMessageId messageFlagged = mailboxProbe.appendMessage(username, new MailboxPath(MailboxConstants.USER_NAMESPACE, username, "mailbox"),
+            new ByteArrayInputStream("Subject: test\r\n\r\ntestmail".getBytes()), new Date(), false, new Flags(Flags.Flag.FLAGGED));
+
+        await();
+
+        given()
+            .header("Authorization", accessToken.serialize())
+            .body("[[\"getMessageList\", {\"filter\":{\"hasKeyword\":[\"$Flagged\"]}}, \"#0\"]]")
```

Comment:
Specs says:

```
hasKeyword: String This message must have the given keyword to match the condition.
notKeyword: String This message must not have the given keyword to match the condition.
```

The parameter of **hasKeyword** and **notKeyword** should be a simple String.